### PR TITLE
Add lots of missing doc comments.

### PR DESCRIPTION
**Pull request by Eilon**
Also did some minor renames to match extension method patterns.

@HaoK @blowdart @Tratcher

### CODE REVIEW

File: src/Microsoft.AspNet.Authentication.Cookies/CookieServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -8,10 +8,16 @@
...
     /// </summary>
     public static class CookieServiceCollectionExtensions
     {
+        /// <summary>
+        /// Adds cookie authentication services to the specified <see cref="IServiceCollection" />. 
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <param name="configure">An action delegate to configure the provided <see cref="CookieAuthenticationOptions"/>.</param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
         public static IServiceCollection AddCookieAuthentication(this IServiceCollection services, Action<CookieAuthenticationOptions> configure)
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Authentication.Cookies/CookieServiceCollectionExtensions.cs]: @HaoK aren't these obsolete?

**Comment by Tratcher** [comment]
:shipit:

File: src/Microsoft.AspNet.Authentication.Cookies/CookieServiceCollectionExtensions.cs (hunk 1 above)
Comment (HaoK) [inline src/Microsoft.AspNet.Authentication.Cookies/CookieServiceCollectionExtensions.cs]: Yeah I will go through and nuke these, they are worse than obsolete, they pretend to do stuff :)

Comment (Eilon) [inline src/Microsoft.AspNet.Authentication.Cookies/CookieServiceCollectionExtensions.cs]: Ok well I'm merged, so you can now delete my comments :smile:

**Comment by Eilon** [comment]
@Tratcher thanks!

**Pull request merged**